src/groebner_basis.jl

```diff
@@ -0,0 +1,90 @@
+
+DP = SymbolicUtils.DynamicPolynomials
+
+# returns the result of promotion of all itr elements types to a single one
+commoneltype(x) = typeof(x)
+commoneltype(arr::AbstractArray)      = commoneltype(x for x in arr)
+commoneltype(itr::Base.ValueIterator) = commoneltype(x for x in itr)
+commoneltype(itr::Base.Generator)     = mapreduce(commoneltype, promote_type, itr, init=Int)
+
+#=
+Converts an array of symbolic polynomials
+into an array of DynamicPolynomial's
+=#
+function symbol_to_poly(sympolys::AbstractArray, check)
+    # find all vars present in system
+    vars = Any[]
+    for expr in sympolys
+        get_variables!(vars, expr)
+    end
+
+    # how to make polynomial_coeffs consider constants as monomials?
+    # e.g. for now it says that res for x + 1 is 1, but we want
+    polyforms, res = polynomial_coeffs(sympolys, vars)
+
+    # check if constant terms do not contain any left symbolic expressions
+    check && any(iswrapped, res) && error("Not a polynomial in input.")
+
+    # discover a common coefficient type
+    coefftype = commoneltype(values(pf) for pf in polyforms)
+    coefftype = promote_type(coefftype, commoneltype(res))
+    @assert coefftype <: Union{Integer, Rational} "Only integer and rational coefficients are supported as input."
+
+    # is there a better criterion for sorting variables?
+    sort!(vars, by=string)
+
+    DP.@polyvar xs[1:length(vars)]
+
+    # sym variable => poly variable
+    sym2poly = Dict(v => x for (v, x) in zip(vars, xs))
+
+    polys = Vector{DP.Polynomial{true, coefftype}}(undef, length(sympolys))
+    for (i, (polyform, r)) in enumerate(zip(polyforms, res))
+        polys[i] = sum(substitute(c*m, sym2poly) for (c, m) in polyform; init=r)
+    end
+
+    polys, sym2poly
```

Comment:
You can try using this:

```julia
using Bijections

pvar2sym, sym2term = Bijection{Any,Any}, Dict{Sym,Any}

polys = map(x->PolyForm(x, pvar2sym, sym2term), xs)
```

`polys[1].p` will be a DynamicPolynomial and so on.

So you can call groebner on `gs = map(x->x.p, polys)` and then convert them back by constructing `PolyForm(gs[1], sym2pvar, pvar2sym)`

Your current function does sorting, but the above technique leaves it to DP to create an ordering.

Reply:
That could be nice! But for some reason I can't reproduce it =(

```julia
using Symbolics, SymbolicUtils, Bijections;
Symbolics.@variables x y;
pvar2sym, sym2term = Bijection{Any,Any}(), Dict{Symbolics.Sym,Any}();
poly = PolyForm(x + y, pvar2sym, sym2term);
dump(poly, maxdepth=2)
```

gives me 
```
Num
  val: SymbolicUtils.Add{Real, Int64, Dict{Any, Number}, Nothing}        
    coeff: Int64 0
    dict: Dict{Any, Number}
    sorted_args_cache: Base.RefValue{Any}
    hash: Base.RefValue{UInt64}
    metadata: Nothing nothing
```

and an empty `pvar2sym` & `sym2term` . Is there something I'm probably missing?

Reply:
You just need to `unwrap` the expression since you're using `@variables`. It wraps the variables and subsequent expressions in `Num`.
```julia
julia> using Symbolics: unwrap

julia> poly = PolyForm(unwrap(x + y), pvar2sym, sym2term);

julia> typeof(poly.p)
DynamicPolynomials.Polynomial{true, Int64}
```